Apps/WebClient/src/ClientApp/src/components/private/sharing/InvitationRecordTypesStepComponent.vue

```diff
@@ -0,0 +1,85 @@
+<script setup lang="ts">
+import useVuelidate from "@vuelidate/core";
+import { minLength, required } from "@vuelidate/validators";
+import { computed, ref } from "vue";
+
+import { EntryType, entryTypeMap } from "@/constants/entryType";
+import { useDelegateStore } from "@/stores/delegate";
+import ConfigUtil from "@/utility/configUtil";
+import ValidationUtil from "@/utility/validationUtil";
+
+defineExpose({ saveStep });
+
+const delegateStore = useDelegateStore();
+
+const selectedRecordTypes = ref<EntryType[]>([]);
+
+let entryTypeOptions: Array<{ title: string; value: EntryType }> = [];
+
+const validations = computed(() => ({
+    selectedRecordTypes: {
+        required,
+        minLength: minLength(1),
+    },
+}));
+
+const recordTypesErrorMessages = computed(() =>
+    ValidationUtil.getErrorMessages(v$.value.selectedRecordTypes)
+);
+
+const v$ = useVuelidate(validations, {
+    selectedRecordTypes,
+});
+
+function saveStep(): void {
+    v$.value.$touch();
+    if (v$.value.$invalid) return;
+    delegateStore.captureInvitationRecordTypes(selectedRecordTypes.value);
+}
+
+// INIT
+entryTypeOptions = ConfigUtil.enabledDatasets()
+    .map((dataset: EntryType) => {
+        const entryDetails = entryTypeMap.get(dataset);
+        return entryDetails === undefined
+            ? undefined
+            : {
+                  title: entryDetails.name,
+                  value: dataset,
+              };
+    })
+    .filter((option) => option !== undefined) as Array<{
+    title: string;
+    value: EntryType;
+}>;
+
+if (delegateStore.invitationWizardState?.recordTypes !== undefined) {
+    selectedRecordTypes.value =
+        delegateStore.invitationWizardState.recordTypes ?? [];
+}
```

Comment:
Combining the undefined check and null-coalescing operator is redundant - one can be dropped.